ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java

```diff
@@ -23,13 +26,13 @@
 
 public final class BeaconBlocksByRangeRequestMessage
     implements RpcRequest, SimpleOffsetSerializable, SSZContainer {
-
   private final UnsignedLong startSlot;
   private final UnsignedLong count;
   private final UnsignedLong step;
 
   public BeaconBlocksByRangeRequestMessage(
       final UnsignedLong startSlot, final UnsignedLong count, final UnsignedLong step) {
+    checkArgument(count.compareTo(UnsignedLong.valueOf(MAX_REQUEST_BLOCKS)) <= 0);
```

Comment:
Can we change this to throw an `RpcException` rather than `IllegalArgumentException`.  Otherwise if we receive a request that fails this check we'll report an internal error back to the peer instead of a bad request.